gtsfm/densify/mvs_patchmatchnet.py

```diff
@@ -0,0 +1,251 @@
+        thresholds: List[float] = [
+            DEFAULT_GEOMETRIC_PIXEL_THRESH,
+            DEFAULT_GEOMETRIC_DEPTH_THRESH,
+            DEFAULT_PHOTOMETRIC_THRESH,
+        ],
+    ) -> np.ndarray:
+        """Get dense point cloud using PatchmatchNet from GtsfmData. The method implements the densify method in MVSBase
+        Ref: Wang et al. https://github.com/FangjinhuaWang/PatchmatchNet/blob/main/eval.py
+
+        Args:
+            images: image dictionary obtained from loaders
+            sfm_result: result of GTSFM after bundle adjustment
+            num_views: number of views, containing 1 reference view and (num_views-1) source views
+                Defaults to DEFAULT_VIEW_NUMBER
+            thresholds: geometric pixel threshold, geometric depth threshold, and photometric
+                threshold for filtering inference results.
+                Defaults to [DEFAULT_GEOMETRIC_PIXEL_THRESH, DEFAULT_GEOMETRIC_DEPTH_THRESH, DEFAULT_PHOTOMETRIC_THRESH]
+                1. for geometric thresholds, small threshold means high accuracy and low completeness
+                2. for photometric thresholds, large threshold means high accuracy and low completeness
+
+        Returns:
+            3D coordinates (in the world frame) of the dense point cloud, (point number, 3)
+        """
+        dataset = PatchmatchNetData(images=images, sfm_result=sfm_result, num_views=num_views)
+        loader = DataLoader(
+            dataset=dataset,
+            batch_size=DEFAULT_BATCH_SIZE,
+            shuffle=False,
+            num_workers=DEFAULT_WORKERS_NUMBER,
+            drop_last=False,
+        )
+
+        model = PatchmatchNet(
+            patchmatch_interval_scale=[0.005, 0.0125, 0.025],
+            propagation_range=[6, 4, 2],
+            patchmatch_iteration=[1, 2, 2],
+            patchmatch_num_sample=[8, 8, 16],
+            propagate_neighbors=[0, 8, 16],
+            evaluate_neighbors=[9, 9, 9],
+        )
+        model = nn.DataParallel(model)
+
+        # Check if cuda devices is supported, and load the pretrained model
+        #   the pretrained checkpoint should be pre-downloaded using gtsfm/download_model_weights.sh
+        if torch.cuda.is_available():
+            model.cuda()
+            state_dict = torch.load(PATCHMATCHNET_WEIGHTS_PATH)
+        else:
+            state_dict = torch.load(PATCHMATCHNET_WEIGHTS_PATH, map_location=torch.device("cpu"))
+
+        model.load_state_dict(state_dict["model"])
+        model.eval()
+
+        depth_est_list = {}
+        confidence_est_list = {}
+        with torch.no_grad():
+            for batch_idx, sample in enumerate(loader):
+                start_time = time.time()
+
+                # Check if cuda devices is supported, and store the inference data to the target device
+                if torch.cuda.is_available():
+                    sample_device = tocuda(sample)
+                else:
+                    sample_device = copy.copy(sample)
```

Comment:
why do we need a copy?

Reply:
I saw you removed the copy, but I just want to confirm that the copy wasnt needed. Could you post a screenshot of the result?